www/js/intro.js

```diff
@@ -134,27 +134,80 @@ angular.module('emission.intro', ['emission.splash.startprefs',
       });
   }
 
-  $scope.login = function() {
-    window.cordova.plugins.BEMJWTAuth.signIn().then(function(userEmail) {
+  $scope.loginExisting = function() {
+    $scope.data = {};
+    const tokenPopup = $ionicPopup.show({
+        template: '<input type="String" ng-model="data.existing_token">',
+        title: 'Enter the existing token that you have',
+        scope: $scope,
+        buttons: [
+          {
+            text: '<b>OK</b>',
+            type: 'button-positive',
+            onTap: function(e) {
+              if (!$scope.data.existing_token) {
+                //don't allow the user to close unless he enters a username
+
+                e.preventDefault();
+              } else {
+                return $scope.data.existing_token;
+              }
+            }
+          },{
+            text: '<b>Cancel</b>',
+            type: 'button-stable',
+            onTap: function(e) {
+              return null;
+            }
+          }
+        ]
+    });
+    tokenPopup.then(function(token) {
+        if (token != null) {
+            $scope.login(token);
+        }
+    }).catch(function(err) {
+        $scope.alertError(err);
+    });
+  };
+
+  $scope.login = function(token) {
+    window.cordova.plugins.BEMJWTAuth.setPromptedAuthToken(token).then(function(userEmail) {
       // ionicToast.show(message, position, stick, time);
       // $scope.next();
       ionicToast.show(userEmail, 'middle', false, 2500);
       if (userEmail == "null" || userEmail == "") {
         $scope.alertError("Invalid login "+userEmail);
       } else {
         CommHelper.registerUser(function(successResult) {
-          UpdateCheck.getChannel().then(function(retVal) {
-            CommHelper.updateUser({
-             client: retVal
+          const uuid = successResult.uuid;
+          return CommHelper.updateUser({ branch: 'rciti1' })
+            .then(function() {
+              const thisUuid = uuid ? uuid : 'undefined';
+              const returnUrl = 'https://emission-app.fourstep.dev/survey-success-static/';
+              console.log('returnUrl', returnUrl);
+              const url = `https://up.fourstep.dev/?returnUrl=${returnUrl}&uuid=${thisUuid}`;
+              const iab = $window.cordova.InAppBrowser.open(url, '_blank');
+              const listener = function(event) {
+                console.log("started loading, event = " + JSON.stringify(event));
+                if (event.url === returnUrl || event.url === 'https://ee.kobotoolbox.org/thanks') {
+                  iab.removeEventListener('loadstart', listener);
+                  iab.close();
+                  ionicToast.show(userEmail, 'middle', false, 2500);
+                  $scope.finish();
+                }
+              };
```

Comment:
This is to auto-close the survey after completion, right?
would be good to put this into `surveylaunch.js`, since:
- others might also want to use it
- it is more likely to survive changes to the onboarding

In general, I would recommend against putting substantial functionality in `intro.js` since that tends to be heavily customized.

Reply:
I can put this bits into `surveylaunch.js`.

Reply:
great, but please make sure it is configurable/modular. Not everybody will use Enketo (the CEO project is using Google Forms for now), and not every enketo user will want to auto-close surveys.

Reply:
LMK the ETA for this change. I'd like to figure out whether I should wait for it or not.

Reply:
Should be done in a few hours.

Reply:
Done